projects/dotcms-field-elements/src/components/dot-checkbox/dot-checkbox.tsx

```diff
@@ -1,58 +1,133 @@
-import { Component, Prop, State, Event, EventEmitter } from '@stencil/core';
-import { generateId, getDotOptionsFromFieldValue } from '../../utils';
-import { DotOption } from '../../models/dot-option.model';
+import { Component, Prop, State, Element, Method } from '@stencil/core';
 import Fragment from 'stencil-fragment';
+import { DotOption, DotFieldStatus } from '../../models';
+import {
+    emitEvent,
+    getClassNames,
+    getOriginalStatus,
+    getTagHint,
+    getTagError,
+    getTagLabel,
+    getErrorClass,
+    getDotOptionsFromFieldValue,
+    updateStatus
+} from '../../utils';
 
 @Component({
     tag: 'dot-checkbox',
     styleUrl: 'dot-checkbox.scss'
 })
 export class DotCheckboxComponent {
+    @Element() el: HTMLElement;
+
+    @Prop() disabled = false;
+    @Prop() name: string;
     @Prop() label: string;
     @Prop() hint: string;
     @Prop() options: string;
-    @Prop() value: string;
-    @Event() onChange: EventEmitter;
+    @Prop() required: boolean;
+    @Prop() requiredmessage: string;
+    @Prop({ mutable: true }) value: string;
 
     @State() _options: DotOption[];
-    @State() _value: string;
-    _label: string;
-    _values = {};
+    @State() _valid = true;
+    @State() status: DotFieldStatus = getOriginalStatus();
+    _dotTouched = false;
+    _dotPristine = true;
 
     componentWillLoad() {
         this._options = getDotOptionsFromFieldValue(this.options);
-        this._label = `dotCheckbox_${generateId()}`;
+        this.emitValueChange();
+        this.emitStatusChange();
     }
 
-    // Todo: find how to set proper TYPE in TS
-    setValue(event): void {
-        const checkBoxVal = event.target.value.toString();
-        // Format values to be emmitted, this might change when implemented on the form
-        this._values = { ...this._values, [checkBoxVal]: !this._values[checkBoxVal] };
-        this.onChange.emit({ value: this._values });
+    hostData() {
+        return {
+            class: getClassNames(this.status, this.isValid())
+        };
+    }
+
+    /**
+     * Reset properties of the field, clear value and emit events.
+     *
+     * @memberof DotSelectComponent
+     */
+    @Method()
+    reset(): void {
+        this.value = '';
+        this.status = getOriginalStatus(this.isValid());
+        this.emitValueChange();
+        this.emitStatusChange();
     }
 
     render() {
         return (
             <Fragment>
```

Comment:
should we add a disable attribute or class, to apply styles to the hole set ( right now only disable the checkboxes ) ?. for example if want to change the color of the labels.

Reply:
I guess we'll added in the future if needed